utilities/Microsoft.Quantum.Katas/CheckKataMagic.cs

```diff
@@ -162,21 +172,29 @@ public virtual bool Simulate(OperationInfo test, string userAnswer, IChannel cha
 
             try
             {
-                var qsim = CreateSimulator();
+                List<SimulatorBase> simulators = CreateSimulators(channel, test);
                 var hasWarnings = false;
 
-                qsim.DisableLogToConsole();
-                qsim.Register(skeletonAnswer.RoslynType, referenceAnswer.RoslynType, typeof(ICallable));
-                qsim.OnLog += (msg) =>
+                foreach(SimulatorBase qsim in simulators)
                 {
-                    hasWarnings = msg?.StartsWith("[WARNING]") ?? hasWarnings;
-                    channel.Stdout(msg);
-                };
+                    Logger.LogDebug($"Simulating test {test.FullName} on {qsim.GetType().Name}");
+                    var simHasWarnings = false;
+
+                    qsim.DisableLogToConsole();
+                    qsim.Register(skeletonAnswer.RoslynType, referenceAnswer.RoslynType, typeof(ICallable));
 
-                var value = test.RunAsync(qsim, null).Result;
+                    qsim.OnLog += (msg) =>
+                    {
+                        simHasWarnings = msg?.StartsWith("[WARNING]") ?? simHasWarnings;
+                        if(simHasWarnings == true)
+                            hasWarnings = true;
+                        channel.Stdout(msg);
+                    };
 
-                if (qsim is IDisposable dis) { dis.Dispose(); }
+                    var value = test.RunAsync(qsim, null).Result;
 
+                    if (qsim is IDisposable dis) { dis.Dispose(); }
+                }
                 return !hasWarnings;
             }
             catch (AggregateException agg)
```

Comment:
@anjbur Did some changes regarding that. 
- First, ensured that `CreateSimulators` should never return an empty list. 
  I initially omitted that check since any problem in simulator creation will throw an exception. But this creates a problem if the simulatorList isn't populated after successful creation of the simulators. This might happen in the present scenario if the simulator created is not of the type `SimulatorBase`. Also, a second check wouldn't hurt much  because the impact of such a bug would be high since no tests could be validated because of this bug.
- Secondly, I also added some simulator specific messages.